src/DocoptNet/Docopt.cs

```diff
@@ -25,6 +25,27 @@ partial class Docopt
         protected IDictionary<string, ValueObject> Apply(string doc, Tokens tokens,
             bool help = true,
             object version = null, bool optionsFirst = false, bool exit = false)
+        {
+            return Apply(doc, tokens, ApplicationResultAccumulators.ValueObjectDictionary, help, version, optionsFirst, exit);
+        }
+
+        internal T Apply<T>(string doc, IApplicationResultAccumulator<T> accumulator)
+        {
+            return Apply(doc, new Tokens(Enumerable.Empty<string>(), typeof (DocoptInputErrorException)), accumulator);
+        }
+
+        internal T Apply<T>(string doc, ICollection<string> argv,
```

Comment:
Do we want this to be public? I can't see how a user would inject an accumulator otherwise.

Reply:
Yes, I think eventually. For now, I wanted to bring this as an internal design, refactoring and changes that can eventually be added to the public API.